Sources/SourceControl/GitRepository.swift

```diff
@@ -702,10 +703,8 @@ public final class GitRepository: Repository, WorkingCheckout {
 
     /// Returns true if a revision exists.
     public func exists(revision: Revision) -> Bool {
-        self.lock.withLock {
-            let output = try? callGit("rev-parse", "--verify", "\(revision.identifier)^{commit}")
-            return output != nil
-        }
+        let output = try? callGit("rev-parse", "--verify", "\(revision.identifier)^{commit}")
+        return output != nil
```

Comment:
Assuming it is intentional that we don't use a lock anymore here, but I am not sure I understand why.